great_expectations/core/expectation_suite.py

```diff
@@ -1110,6 +1111,16 @@ def render(self) -> None:
                 )
             )
 
+    def serialize(self) -> _IdentifierBundle:
```

Comment:
Moved any encoding logic to the actual class being serialized (move from `_encode_suite` in validation config to `serialize` as instance method on suite).

I don't love exposing these publicly but how do we get around this? Do we want custom serializer classes to encapsulate this logic?

Reply:
This only serializes this object when it is used in another object right? I would call this `serialize_identifier_bundle()` or just `identifier_bundler()`. This name should be uniform across any object that needs to do this. I think this public since whatever object serializes this expects this method to exist. We could formalize it as an internal protocol.

Reply:
I agree for suites and validation definitions but renderers and batch definitions have different outputs (`_EncodedValidationData` and `class_name/module_name`, respectively).